pkg/apis/certificates/validation/validation.go

```diff
@@ -50,59 +52,53 @@ func ValidateCertificateRequestName(name string, prefix bool) []string {
+		// <resource-type-name>.<fully-qualified-group-name>/<resource-namespace>.<resource-name>.
+		// The max length of a resource type name is 63 characters (DNS1035Label max length)
+		// The max length of a group name is 253 characters (DNS1123Subdomain max length)
+		// The max length of a namespace name is 63 characters (DNS1123Label max length)
+		// The max length of a resource name is 253 characters (DNS1123Subdomain max length)
+		// We then add an additional 3 characters to account for the two '.' and one '/'.
+		signerNameMaxLength := 635
+		if len(csr.Spec.SignerName) > signerNameMaxLength {
+			allErrs = append(allErrs, field.TooLong(specPath.Child("signerName"), csr.Spec.SignerName, signerNameMaxLength))
 		} else {
-			domain := parts[0]
-			path := parts[1]
-			signerName := *csr.Spec.SignerName
-
-			// ensure the path portion has a length greater than 0
-			if len(domain) == 0 {
-				allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), signerName, "domain component must be specified"))
-			}
-
-			// check the domain portion to ensure it consists of valid dns labels
-			domainOnDots := strings.Split(domain, ".")
-			for _, s := range domainOnDots {
-				if errs := validation.NameIsDNSLabel(s, false); len(errs) > 0 {
-					for _, err := range errs {
-						allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), signerName, err))
-					}
-				}
-			}
-
-			// ensure the path portion has a length greater than 0
-			if len(path) == 0 {
-				allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), signerName, "path component must be specified"))
-			}
-
-			// check the path portion to ensure it consists of valid dns labels
-			pathOnDots := strings.Split(domain, ".")
-			for _, s := range pathOnDots {
-				if errs := validation.NameIsDNSLabel(s, false); len(errs) > 0 {
-					for _, err := range errs {
-						allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), signerName, err))
+			parts := strings.Split(csr.Spec.SignerName, "/")
+			if len(parts) != 2 {
+				allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), csr.Spec.SignerName, "must be a fully qualified domain and path of the form 'example.com/signer-name'"))
+			} else {
+				domain := parts[0]
+				path := parts[1]
+				signerName := csr.Spec.SignerName
+
+				// check the domain portion to ensure it is a valid fully qualified domain name
+				allErrs = append(allErrs, utilvalidation.IsFullyQualifiedDomainName(specPath.Child("signerName"), domain)...)
+
+				// ensure the path portion has a length greater than 0
+				if len(path) == 0 {
+					allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), signerName, "path component must be specified"))
+				} else {
+					// check the path portion to ensure it consists of valid dns labels
+					pathOnDots := strings.Split(path, ".")
+					for _, s := range pathOnDots {
+						if errs := validation.NameIsDNSLabel(s, false); len(errs) > 0 {
```

Comment:
You do not need this `len` check.

`for _, err := range validation.NameIsDNSLabel(s, false) {`